var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -342,6 +347,15 @@ def on_off(varstr):
                 % on_off('^superlu-dist+openmp')
             ])
 
+
+        # Building with Trilinos
+        if '+trilinos' in spec:
+            args.extend([
+                '-DTrilinos_ENABLE=ON',
```

Comment:
Can you explicitly disable this in an else clause?

Reply:
I can, although Ill note we do not do this for other third-party libraries. Is this the preferred pattern? If so, I will add else clauses for the other third-party libraries as well.

Reply:
Yes, this is the preferred pattern. CMake (and other build systems) like to change the defaults of variants based on whether or not the dependencies are found, leading to non-deterministic builds with missing dependencies. We like to always explicitly disable variants to prevent this.

Reply:
OK I have gone through and added an else clause in the relevant places.